### ISSUE TITLE
Error response from daemon: manifest for ethorbit/discord_sticky-message-bot:latest not found

### ISSUE THREAD

**Issue by bxpana** [Ethorbit/Discord_Sticky-Message-Bot#7]
Tried to run via docker (`docker pull ethorbit/discord_sticky-message-bot`) and got the following error message
```
Error response from daemon: manifest for ethorbit/discord_sticky-message-bot:latest not found: manifest unknown: manifest unknown
```

Is there something I'm missing?

**Comment by Ethorbit** [comment]
> Tried to run via docker (`docker pull ethorbit/discord_sticky-message-bot`) and got the following error message
> 
> ```
> Error response from daemon: manifest for ethorbit/discord_sticky-message-bot:latest not found: manifest unknown: manifest unknown
> ```
> 
> Is there something I'm missing?

Hi, for whatever reason the only tag that exists is :linux 
so you'll have to do `docker pull ethorbit/discord_sticky-message-bot:linux`

I'll try to change that

**Comment by bxpana** [comment]
@Ethorbit thank you! That worked but when I tried to run the bot it looks like it's not meant for `arm`

Is there another way that I could host this instead of through Docker?

**Comment by Ethorbit** [comment]
> @Ethorbit thank you! That worked but when I tried to run the bot it looks like it's not meant for `arm`
> 
> Is there another way that I could host this instead of through Docker?

Install nodejs, git and `git clone https://github.com/Ethorbit/Discord_Sticky-Message-Bot.git` 
Create a .env file in ./Discord_Sticky-Message-Bot/
Add this to the .env file:
```
BOT_TOKEN=YOURBOTsTOKEN
STICKY_COOLDOWN=20000
```

Test the bot to make sure it works: `node ./Discord_Sticky-Message-Bot/src/bot.js`
If it turns on, you can make it run as soon as the system starts. This can be done quite a few ways, I personally use systemd.

That's pretty much what the container does. You can also try building the included Dockerfile on arm and see if it works. If it works, please let me know so I can add support for it as I currently have no arm system to test on.

**Comment by bxpana** [comment]
I'm still working on this and have had to make some adjustments to the code (add `intents` for new discord.js version), but I can't get it to run. I run `node ./Discord_Sticky-Message-Bot/src/bot.js` but I get 
```
if (!token || typeof token !== 'string') throw new Error('TOKEN_INVALID');
                                                   ^

Error [TOKEN_INVALID]: An invalid token was provided.
```

I've changed the token twice and have tried as a string and not being a string. The `.env` file is in the `Discord_Sticky-Message-Bot` directory too

**Comment by Ethorbit** [comment]
> I'm still working on this and have had to make some adjustments to the code (add `intents` for new discord.js version), but I can't get it to run. I run `node ./Discord_Sticky-Message-Bot/src/bot.js` but I get
> 
> ```
> if (!token || typeof token !== 'string') throw new Error('TOKEN_INVALID');
>                                                    ^
> 
> Error [TOKEN_INVALID]: An invalid token was provided.
> ```
> 
> I've changed the token twice and have tried as a string and not being a string. The `.env` file is in the `Discord_Sticky-Message-Bot` directory too

I believe the .env might need to be placed inside the src/ folder where bot.js resides for it to work

**Comment by bxpana** [comment]
Tried that and it didn't work either. Tried running with a config.json instead of .env and using `dotenv`. I got the bot online, but it doesn't react to any of the commands.
I run `node bot.js` from the `src` directory
Terminal hangs
Bot is online
No response to `!sticky add <channel ID> <message>` 


Do you have a working example of it?

**Comment by Ethorbit** [comment]
My best guess is that it's throwing an error because BOT_TOKEN=whatever isn't a string, but BOT_TOKEN="whatever" is. Maybe putting quotes around it will get it working.

I'll report back after testing for both Windows and Linux tomorrow, and will be sure to make the code convert BOT_TOKEN to a string instead of checking its type to prevent future issues like this.

Make sure the bot has these permissions when adding to a server, so it can see your commands:
* Read Message History
* Send Messages
* Manage Messages

**Comment by Ethorbit** [comment]
> Tried that and it didn't work either. Tried running with a config.json instead of .env and using `dotenv`. I got the bot online, but it doesn't react to any of the commands. I run `node bot.js` from the `src` directory Terminal hangs Bot is online No response to `!sticky add <channel ID> <message>`
> 
> Do you have a working example of it?

It turns out the bot was looking for the .env file relative to *the directory the terminal was* in rather than *to the bot.js file itself*, this explains why it wasn't able to find it. This issue has been fixed, it will now recurse directories until it finds one with a .env file inside of it. 

As for the bot not responding to commands, it was due to the Discord developers making a (dumb) change which requires all bots to have the "Message Content Intent" option enabled in the Bot tab of the Application or else they will see every message as empty and thus commands won't work.

I have created a [Wiki](https://github.com/Ethorbit/Discord_Sticky-Message-Bot/wiki) for this git repository which covers discord bot creation / inviting, and installing / running the bot.

Hope this solves all problems.
